### PR TITLE
Add test methods for zero, real, diagonal, symmetric and hermitian matrices

### PR DESCRIPTION
**Pull request by rikardn**
Add methods `is_zero`, `is_real`, `is_diagonal`, `is_symmetric` and `is_hermitian` to `DenseMatrix`.

This PR depends on #1702.

### CODE REVIEW

File: symengine/basic.h [hunk 1]
```diff
@@ -227,6 +227,43 @@ inline bool is_true(tribool x)
     return x == tribool::tritrue;
 };
 
+inline bool is_false(tribool x)
```
Comment (isuruf) [inline symengine/basic.h:230]: Can you keep the declaration here and move the implementation to `basic-inl.h`? Sorry, I forgot to mention this in `is_true`.

**Review by isuruf — COMMENTED** [review]
Looks good to me except for one small nitpick.

File: symengine/basic.h (hunk 1 above)
Comment (rikardn) [reply symengine/basic.h]: Ok!

**Comment by rikardn** [comment]
I have moved the implementations of the tribool functions.

It seems as if this PR also contains the commits from PR #1702. Perhaps it is enough to merge this one.

**Comment by isuruf** [comment]
Thanks

**Comment by rikardn** [comment]
Thanks for reviewing!
